blues/simulation.py

```diff
@@ -479,43 +469,41 @@ def simulateNCMC(self, nstepsNC=5000, ncmc_traj=None,
         self.move_engine.selectMove()
         move_idx = self.move_engine.selected_move
         move_name = self.move_engine.moves[move_idx].__class__.__name__
+
         for nc_step in range(int(nstepsNC)):
             start = time.time()
-            self._initialSimulationTime = self.nc_context.getState().getTime()
+            self._initialSimulationTime = self.nc_sim.context.getState().getTime()
+
             try:
                 #Attempt anything related to the move before protocol is performed
                 if nc_step == 0:
-                    self.nc_context = self.move_engine.moves[self.move_engine.selected_move].beforeMove(self.nc_context)
+                    self.nc_sim.context = self.move_engine.moves[self.move_engine.selected_move].beforeMove(self.nc_sim.context)
+
                 # Attempt selected MoveEngine Move at the halfway point
                 #to ensure protocol is symmetric
                 if self.movestep == nc_step:
                     #Do move
                     self.log.info('Performing %s...' % move_name)
-                    self.nc_context = self.move_engine.runEngine(self.nc_context)
+                    self.nc_sim.context = self.move_engine.runEngine(self.nc_sim.context)
 
                 # Do 1 NCMC step with the integrator
-                self.nc_integrator.step(1)
+                self.nc_sim.step(1)
```

Comment:
This should be functionally the same. The only issue I can think of is that if you attach a reporter that has a !=1 reporting interval, the reporting interval can be messed up if a nan occurs in a given iteration, since the iteration will be stopped partway. So you could put a `self.nc_sim.currentStep = 0` in the except clause that catches these to get around that.

Reply:
I think you still need to include this?

Reply:
If the iteration hits a NaN, the process still calls the `acceptRejectNCMC` function which resets the currentStep. See: https://github.com/MobleyLab/blues/pull/116/files/436749a83e2c2279536d844f0d3f992ad4f34d17#diff-3c46bef0e4bdea1cebbb3cbc654c214bR460 

Does that look okay to you?